.github/workflows/validate-crd.yaml

```diff
@@ -0,0 +1,48 @@
+name: Check CRD version update
+
+on:
+  pull_request:
+    paths:
+      - 'pkg/k8s/apis/cilium.io/client/crds/v1alpha1/*.yaml'
+      - 'pkg/k8s/apis/cilium.io/v1alpha1/register.go'
+
+jobs:
+  check-version:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check for CRD changes and version update
+        run: |
+          crd_changed=0
+          version_changed=0
+
+          # Check for CRD changes
+          crd_changes=$(git diff --name-only ${{ github.base_ref }} ${{ github.head_ref }} -- pkg/k8s/apis/cilium.io/client/crds/v1alpha1/*.yaml)
+          if [ -n "$crd_changes" ]; then
+            crd_changed=1
+          fi
+
+          # Check for version variable changes
+
+          old_version=$(git show ${{ github.event.pull_request.base.sha }}:pkg/k8s/apis/cilium.io/v1alpha1/register.go | grep 'CustomResourceDefinitionSchemaVersion' | awk -F'"' '{print $2}')
+          new_version=$(git show HEAD:pkg/k8s/apis/cilium.io/v1alpha1/register.go | grep 'CustomResourceDefinitionSchemaVersion' | awk -F'"' '{print $2}')
+
+          echo "old_version=$old_version"
+          echo "new_version=$new_version"
+
+          if [ "$old_version" != "$new_version" ]; then
+            version_changed=1
+          fi
+
+          if [ "$crd_changed" -eq 1 ] && [ "$version_changed" -eq 0 ]; then
+            echo "CRD changed but version not updated"
+            exit 1
+          fi
+
+          if [ "$crd_changed" -eq 0 ] && [ "$version_changed" -eq 1 ]; then
+            echo "Version updated but CRD not changed"
```

Comment:
Similar comment as above - could you write more precisely what's wrong, i.e. point the user to concrete files?